core/src/main/java/org/kiwix/kiwixmobile/core/page/notes/viewmodel/effects/ShowOpenNoteDialog.kt

```diff
@@ -51,35 +54,37 @@ data class ShowOpenNoteDialog(
       ShowNoteDialog,
       { effects.offer(OpenPage(page, zimReaderContainer)) },
       {
-        val item = page as NoteListItem
-        // Check if zimFilePath is not null, and then set it in zimReaderContainer.
-        // For custom apps, we are currently using fileDescriptor, and they only have a single file in them,
-        // which is already set in zimReaderContainer, so there's no need to set it again.
-        item.zimFilePath?.let {
-          val currentZimFilePath = zimReaderContainer.zimCanonicalPath
-          val file = File(it)
-          zimReaderContainer.setZimFile(file)
-          if (zimReaderContainer.zimCanonicalPath != currentZimFilePath) {
-            // if current zim file is not the same set the main page of that zim file
-            // so that when we go back it properly loads the article, and do nothing if the
-            // zim file is same because there might be multiple tabs opened.
-            val settings = activity.getSharedPreferences(
-              SharedPreferenceUtil.PREF_KIWIX_MOBILE,
-              0
-            )
-            val editor = settings.edit()
-            val urls = JSONArray()
-            val positions = JSONArray()
-            urls.put(CONTENT_PREFIX + zimReaderContainer.mainPage)
-            positions.put(0)
-            editor.putString(TAG_CURRENT_FILE, zimReaderContainer.zimCanonicalPath)
-            editor.putString(TAG_CURRENT_ARTICLES, "$urls")
-            editor.putString(TAG_CURRENT_POSITIONS, "$positions")
-            editor.putInt(TAG_CURRENT_TAB, 0)
-            editor.apply()
+        dialogScope.launch {
```

Comment:
There is also a delay in showing the notes dialog, if we open the note from `Notes Screen`(if note's zim file is not currently set in the `zimReaderContainer`).